xdsl/dialects/stim/stim_parser.py

```diff
@@ -0,0 +1,275 @@
+import re
+from collections.abc import Callable, Sequence
+from typing import TypeVar
+
+from xdsl.dialects.builtin import ArrayAttr, FloatData, IntAttr
+from xdsl.dialects.stim import (
+    QubitCoordsOp,
+)
+from xdsl.dialects.stim.ops import QubitAttr, QubitMappingAttr, StimCircuitOp
+from xdsl.ir import Block, Operation, Region
+from xdsl.utils.lexer import Input, Position
+from xdsl.utils.str_enum import StrEnum
+
+T = TypeVar("T")
+
+
+class StimParseError(Exception):
+    position: Position
+    message: str
+
+    def __init__(self, position: Position, message: str) -> None:
+        self.position = position
+        self.message = message
+        super().__init__(f"StimParseError at {self.position}: {self.message}")
+
+
+class Instruction(StrEnum):
+    """
+    Enum for the parse-able instructions in Stim.
+    """
+
+    COORD = "QUBIT_COORDS"
+
+
+NEWLINE = re.compile(r"\n")
+NOT_NEWLINE = re.compile(r"[^\n]*")
+INDENT = re.compile(r"[ \t]*")
+SPACE = re.compile(r"[ \t]")
+ARG = re.compile(r"\d+(\.\d+)?")
+# PAULI = re.compile("|".join(re.re.escape(p.value) for p in PauliSpelling))

```

Comment:
commented code, seems like these might be better merged together with the code that leverages them?